Added code package (Visual Studio), and fix gtk_doc

JL2210 opened the pull request:
Fixes #
Ready this time.
All dependencies satisfied.
Great with sommelier, not so much with just wayland.

uberhacker commented:
Where is libsecret?  That is a dependency I needed.  I will push up a PR soon.

JL2210 commented:
Strange. What does it say?

uberhacker commented on packages/code.rb:
```diff
@@ -0,0 +1,49 @@
+  })
+
+  depends_on 'sommelier'
+  depends_on 'xterm'
```
Why do we need xterm?

JL2210 replied:
It doesn't work without. For some reason, it has to be executed with xterm.

uberhacker replied:
We don't need xterm.  Try this: `$ /usr/local/lib/code/bin/code`.

uberhacker commented on packages/code.rb:
```diff
@@ -0,0 +1,49 @@
+      system "cp", "-rpa", ".", "#{CREW_DEST_PREFIX}/lib/code/"
+      system "echo '#!/bin/bash' > #{CREW_DEST_PREFIX}/bin/code"
+      system "echo >> #{CREW_DEST_PREFIX}/bin/code"
+      system "echo 'xterm -e #{CREW_PREFIX}/lib/code/bin/code' >> #{CREW_DEST_PREFIX}/bin/code"
```
The code executable should work without this crazy xterm stuff.  Where did you get these installation instructions?

JL2210 replied:
Googling stuff.

uberhacker commented on packages/gtk_doc.rb:
```diff
@@ -31,14 +31,17 @@ class Gtk_doc < Package
-    system './autogen.sh --with-xml-catalog=/usr/local/etc/xml/catalog'
-    system './configure --with-xml-catalog=/usr/local/etc/xml/catalog'
+    system "./autogen.sh"
+    system "./configure --with-xml-catalog=/usr/local/etc/xml/catalog --prefix=#{CREW_PREFIX} --libdir=#{CREW_LIB_PREFIX}"
```
```
- --with-xml-catalog=/usr/local/etc/xml/catalog
+ --with-xml-catalog=#{CREW_PREFIX}/etc/xml/catalog
```
You should get in the habit of replacing all hardcoded '/usr/local`s with `#{CREW_PREFIX}` instead.

JL2210 replied:
I didn't do it.

JL2210 replied:
Got it anyway.

uberhacker commented on packages/gtk_doc.rb:
```diff
@@ -31,14 +31,17 @@ class Gtk_doc < Package
+
+  def self.postinstall
+    system "pip3 install --upgrade six --prefix #{CREW_PREFIX}"
+    system "pip2 install --upgrade six --prefix #{CREW_PREFIX}"
```
Why do we need both sixes?

JL2210 replied:
Python 2 and 3

uberhacker replied:
We should only need one six module and it should be in the install section, not postinstall.  I know you didn't make these changes initially but as long as we're "fixing" this package...

JL2210 replied:
OK. Python 2.

JL2210 replied:
Just a warning, gtk_doc fails to build because of XSL stylesheets. Help?

uberhacker replied:
I've run across that problem also.  I just removed the docs with `--disable-gtk-doc` in other packages that had this issue.   Not sure what to do in this case.  Maybe try updating the libxslt package?

uberhacker replied:
I have an idea.  Why don't you just remove this package from this PR and we can deal with it in a separate PR?  This doesn't affect the code package AFAICT.

uberhacker replied:
Also, the pre-built binaries need to be removed. :)

JL2210 replied:
What does AFAICT mean?

JL2210 replied:
No. That wouldn't work. I'll just revert gtk_doc, add a package called six that it depends on.

uberhacker replied:
AFAICT = as far as I can tell

uberhacker commented on packages/libgconf.rb:
```diff
@@ -0,0 +1,25 @@
+  description 'GConf is a system for storing application preferences.'
+  homepage 'https://projects-old.gnome.org/gconf/'
+  version '3.2.6'
+  source_url 'ftp://ftp.gnome.org/pub/GNOME/sources/GConf/3.2/GConf-3.2.6.tar.xz'
```
ftp is evil:
```
- ftp://ftp.gnome.org/pub/GNOME/sources/GConf/3.2/GConf-3.2.6.tar.xz
+ https://ftp.gnome.org/pub/GNOME/sources/GConf/3.2/GConf-3.2.6.tar.xz
```

uberhacker commented on packages/dbus_glib.rb:
```diff
@@ -0,0 +1,25 @@
+  binary_sha256 ({
+  })
+
+  depends_on 'gtk_doc'
```
You are missing `dbus` and  `glib` dependencies.  Test on a fresh crew install.

JL2210 replied:
OK.

JL2210 replied:
LOL, that makes sense.

uberhacker replied:
Remove the `depends_on 'gtk_doc'` dependency?

JL2210 replied:
OK.

uberhacker replied:
Can we not just remove the gtk_doc dependency here and submit the docbook madness in another PR please?

JL2210 replied:
I can't untrack them now. There will be merge conflicts.

JL2210 commented:
Might need a reboot to work.

JL2210 commented:
This is awesome! It highlights your syntax errors and even gives you suggestions on how to correct them!

JL2210 commented:
Try it!

JL2210 commented:
[screenshot 2018-08-08 at 8 24 04 pm](https://user-images.githubusercontent.com/38843657/43871466-c8f3b694-9b4a-11e8-8fd5-d205415a8029.png)

[hello.c.txt](https://github.com/skycocker/chromebrew/files/2272443/hello.c.txt)

superloach commented:
This looks awesome, can't wait! For those who prefer it, what do you think of getting Atom in here?
I tried earlier, but my knowledge of build tools is seriously underpowered. Need to work on that. :/

uberhacker commented on packages/code.rb:
```diff
@@ -0,0 +1,57 @@
+  def self.postinstall
+    puts
+    puts 'Congratulations! You have installed Visual Studio Code on Chrome OS!'.lightgreen
+    puts 'Now, please reboot to restart the Sommelier Daemon, and run \'code\' to start Visual Studio.'.lightgreen
```
Actually, you don't need to go to the extreme of rebooting to start the sommelier "daemon".  All you need to do is exit and re-enter your crosh session.

JL2210 replied:
No, the daemon is run in the background. It stays up, in my experience. Also, the shell still hangs when you try to exit the yellow crosh shell.

uberhacker replied:
Do you have more than one crosh session open?  That maybe the reason.  Besides, rebooting is totally not necessary to start sommelier.  It starts when you login to your shell session from crosh.

JL2210 replied:
The sommelier daemon runs totally in the background. No. This is null anyway, now that I reverted that.

uberhacker replied:
Why do we need to restart?  We only need to start, correct?  A reboot should not be necessary so can you please change the wording?  Maybe just remove and simply say: `Run 'code' to start`?  If they followed the postinstall message, sommelier should be running aready at this point.

JL2210 replied:
Alright. I'll change that. I didn't want a warning to be put out the first time somebody started VS Code.

uberhacker commented on packages/code.rb:
```diff
@@ -0,0 +1,57 @@
+
+  def self.install
+    case ARCH
+    when 'armv7l', 'aarch64'
```
It would be better to have conditional logic that identifies the architecture that is supported rather than unsupported.  So something like:
````
case ARCH
when 'i686', 'x86_64'
  ...
else
  puts
  puts "Visual Studio Code does not currently support your architecture.".lightred
  ...
end
```

uberhacker commented on packages/code.rb:
```diff
@@ -0,0 +1,57 @@
+  depends_on 'sommelier'
+  depends_on 'xterm'
+  depends_on 'libgconf'
+  depends_on 'xdg_base'
```
Why do we need `xdg_base`?  Are you storing settings in `~/.config`?  We should probably list `depends_on 'sommelier'` last since we don't want users to miss the postinstall message.

JL2210 replied:
It fails with a JavaScript error if ~/.config doesn't exist, and this package is a good way to provide that. Not terribly unreasonable.

uberhacker commented on packages/code.rb:
```diff
@@ -0,0 +1,57 @@
+    else
+      system "mkdir", "-p", "#{CREW_DEST_PREFIX}/lib/code"
+      system "mkdir", "-p", "#{CREW_DEST_PREFIX}/bin"
+      system "cp", "-rpa", ".", "#{CREW_DEST_PREFIX}/lib/code/"
```
Can we use `#{CREW_DEST_PREFIX}/share/code` instead?

JL2210 replied:
OK.

JL2210 commented:
Tried that. Has to be run directly, and xterm -e works for that.

On Thu, Aug 9, 2018, 9:41 AM Ed Reel <notifications@github.com> wrote:

> *@uberhacker* commented on this pull request.
> ------------------------------
>
> In packages/code.rb
> <https://github.com/skycocker/chromebrew/pull/2529#discussion_r208931923>:
>
> > +    source_sha256 '5856bbebf38aa05d584da4722869bbe507cf123f69f7ffab5f1532d73dbf3438'
> +  when "i686"
> +    source_url 'https://go.microsoft.com/fwlink/?LinkID=620885'
> +    source_sha256 'af6adc2e2500e50bfebe7ee7b97d661b6e774a590136bf5f89334132a5b292e2'
> +  else
> +    source_url 'https://raw.githubusercontent.com/Microsoft/vscode/1.25.1/README.md'
> +    source_sha256 'c1c5e6ec903730a4e116c1cfd83bb29acd227195d97f2ec8944452808232c310'
> +  end
> +
> +  binary_url ({
> +  })
> +  binary_sha256 ({
> +  })
> +
> +  depends_on 'sommelier'
> +  depends_on 'xterm'
>
> We don't need xterm. Try this: $ /usr/local/lib/code/bin/code.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/skycocker/chromebrew/pull/2529#discussion_r208931923>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AlC1Cb2AMIwtxNKvBZlQ7L1Bu9Wm1i5Bks5uPDv6gaJpZM4V0uCP>
> .
>

uberhacker commented:
It worked fine for me on a fresh crew install after fixing the errors from above.  Again, please test your PR before submitting.  It will save me a lot of effort.

JL2210 commented:
I did. Uncommitted changes.

JL2210 commented on packages/libgconf.rb:
```diff
@@ -0,0 +1,26 @@
+  })
+
+  depends_on 'dbus_glib'
+  depends_on 'gtk2'
```
I noticed it saying it needed gtk2 in the build process, so I stuck it in there. It works, but I'll change that. It doesn't really have anything to do with the version.

JL2210 commented:
Try upgrading gtk_doc for the issue with libgconf failing to install.

JL2210 commented:
Sure. Sorry about that, but it worked for me, even after a fresh crew.

On Thu, Aug 9, 2018, 9:44 AM Ed Reel <notifications@github.com> wrote:

> It worked fine for me on a fresh crew install after fixing the errors from
> above. Again, please test your PR before submitting. It will save me a lot
> of effort.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/skycocker/chromebrew/pull/2529#issuecomment-411762232>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AlC1CQ-A0S2WPaXFdfx8stPbzK25Io4rks5uPDzOgaJpZM4V0uCP>
> .
>

JL2210 commented:
How's it work now?

ukscone commented:
@JL2210 re: crosh hanging when exiting. not 100% sure why but after I added the little /tmp/.X11-unix & pgrep hack to my .bashrc I haven't had crosh hang when exiting since

JL2210 commented:
Nice. I will add that if that fixes the issue.

On Thu, Aug 9, 2018, 11:05 AM Russell Davis <notifications@github.com>
wrote:

> @JL2210 <https://github.com/JL2210> re: crosh hanging when exiting. not
> 100% sure why but after I added the little /tmp/.X11-unix & pgrep hack to
> my .bashrc I haven't had crosh hang when exiting since
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/skycocker/chromebrew/pull/2529#issuecomment-411790380>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AlC1CR3_elk6NVouKFA0aOANJDX7nGThks5uPE_EgaJpZM4V0uCP>
> .
>

ukscone commented:
YMMV and I bet now i've said it it will stop doing it but can't hurt to try :)

JL2210 commented:
VMMV? Where was it that you showed me the .X11.unix hack?

On Thu, Aug 9, 2018, 11:09 AM Russell Davis <notifications@github.com>
wrote:

> YMMV and I bet now i've said it it will stop doing it but can't hurt to
> try :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/skycocker/chromebrew/pull/2529#issuecomment-411791700>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AlC1CYNWgV5by-0UN6oU14rOsupAO6_2ks5uPFCpgaJpZM4V0uCP>
> .
>

uberhacker commented:
@JL2210: I shouldn't need to upgrade one package to fix another.  Each package should work independently of the other.  With that said, please add the six pip module to libgconf in case other packages need to use this as a dependency.

ukscone commented:
@JL2210 can't remember/find the post but here it is again

I put it after the settings needed for sommelier e.g. after the DISPLAY setting

-------
export DISPLAY=:0
if [ ! -d /tmp/.X11-unix ]; then
    mkdir /tmp/.X11-unix
    chmod 1777 /tmp/.X11-unix
    sudo chown root:root /tmp/.X11-unix
fi
if ! pgrep -x "sommelier" > /dev/null
then 
 /usr/local/sbin/sommelierd &
fi

ukscone commented:
n/m as i said it has suddenly started hanging again. smh

JL2210 commented:
No, it works. Thanks!

JL2210 commented:
@uberhacker, Six is a dependency of gtk_doc. It was in the package before, but it was in the build section. I just moved it to the postinstall, added the prefix and changed /usr/local to CREW_PREFIX.

JL2210 commented:
I also bumped the version number.

JL2210 commented:
If you don't mind, I'd really like to fix sommelier here, because the hanging thing is just unbearable. Notice the chmod -R 1777.

ukscone commented:
@JL2210 it seems to mostly work. there must be an edgecase related with how you close the prog. seems to be consist if you close via a menu option but closing via the corner X seems a toss

JL2210 commented:
Yeah. Don't care. I hated having to manually close the Chrome tab and go through the warning each time because it hung.

uberhacker commented on packages/libgconf.rb:
```diff
@@ -0,0 +1,27 @@
+
+  depends_on 'dbus_glib'
+  depends_on 'gtk3'
+  depends_on 'gtk2'
```
This dependency is not needed here.  The gtk2 dependency is needed for code, not libgconf.

JL2210 replied:
I think libgconf has some gtk2 bindings, too. I looked at the configure log.

uberhacker replied:
It looks for gtk3, not gtk2.

JL2210 replied:
OK. Will change.

Ewpratten commented:
How are you doing the gui stuff? crostini? crouton?

uberhacker commented on packages/sommelier.rb:
```diff
@@ -3,7 +3,7 @@
 class Sommelier < Package
```
Please add this change as a separate PR.  We need to keep these things separate because we don't have the binaries yet.  With that said, you will also need to remove the pre-built binaries.

JL2210 replied:
No, I won't. I just changed the postinstall.

JL2210 replied:
Just like with gtk_doc. No difference in the files.

uberhacker replied:
Actually that is correct but I still want a separate PR. :)

JL2210 replied:
I will revert sommelier in this one, and after this is merged I will create a new one.

JL2210 commented:
sommeleir.

uberhacker commented on packages/sommelier.rb:
```diff
@@ -48,14 +48,22 @@ def self.install
+    puts "echo 'sudo chown root:root /tmp/.X11-unix' >> ~/.bashrc".lightblue
+    puts "echo 'if ! pgrep -x \"sommelier\" > /dev/null' >> ~/.bashrc".lightblue
+    puts "echo 'then' >> ~/.bashrc".lightblue
+    puts "echo '/usr/local/sbin/sommelierd &' >> ~/.bashrc".lightblue
```
Please change `/usr/local` to `#{CREW_PREFIX}`.

JL2210 replied:
K

JL2210 commented:
An X forwarding mechanism for Wayland.

uberhacker commented on packages/code.rb:
```diff
@@ -0,0 +1,54 @@
+      puts
+      puts 'Visual Studio Code is currently not supported on ARM and AArch64.'.lightred
+      puts 'Please try HeadMelted.'.lightred
+      puts 'https://code.headmelted.com'.lihghtred
```
lihghtred is misspelled.  Obviously, this hasn't been tested.

JL2210 commented:
I'll just disable gtk docs. That seems ok.

On Thu, Aug 9, 2018, 12:19 PM Ed Reel <notifications@github.com> wrote:

> *@uberhacker* commented on this pull request.
> ------------------------------
>
> In packages/gtk_doc.rb
> <https://github.com/skycocker/chromebrew/pull/2529#discussion_r208993272>:
>
> >      system 'make'
>    end
>
>    def self.install
>      system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
>    end
> +
> +  def self.postinstall
> +    system "pip3 install --upgrade six --prefix #{CREW_PREFIX}"
> +    system "pip2 install --upgrade six --prefix #{CREW_PREFIX}"
>
> I've run across that problem also. I just removed the docs with
> --disable-gtk-doc in other packages that had this issue. Not sure what to
> do in this case. Maybe try updating the libxslt package?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/skycocker/chromebrew/pull/2529#discussion_r208993272>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AlC1CfB_ropekDiosXUo0qTbHvn-9fQ4ks5uPGEfgaJpZM4V0uCP>
> .
>

uberhacker commented on packages/code.rb:
```diff
@@ -0,0 +1,55 @@
+
+  depends_on 'gtk2'
+  depends_on 'libgconf'
+  depends_on 'xdg_base'
```
Why do we need xdg_base?

JL2210 replied:
~/.config

uberhacker commented on packages/code.rb:
```diff
@@ -0,0 +1,55 @@
+      puts
+      puts 'Happy coding!'.lightgreen
+      puts
+      return 1
```
This is not bash, it's ruby so I'm not sure this is valid.

JL2210 replied:
How would you do it? I haven't caught on with all the functions, yet.

uberhacker replied:
How about just remove the line? :)

JL2210 replied:
OK. What would I do to make it fail then?

uberhacker replied:
You could add `abort` but I'm thinking it might not even be necessary.  Test it out a bit and see what happens.

JL2210 commented:
Can I have a minute to work this out in my head?

On Thu, Aug 9, 2018, 12:30 PM Ed Reel <notifications@github.com> wrote:

> *@uberhacker* commented on this pull request.
> ------------------------------
>
> In packages/code.rb
> <https://github.com/skycocker/chromebrew/pull/2529#discussion_r208996454>:
>
> > +  when 'i686'
> +    source_url 'https://go.microsoft.com/fwlink/?LinkID=620885'
> +    source_sha256 'af6adc2e2500e50bfebe7ee7b97d661b6e774a590136bf5f89334132a5b292e2'
> +  else
> +    source_url 'https://raw.githubusercontent.com/Microsoft/vscode/1.25.1/README.md'
> +    source_sha256 'c1c5e6ec903730a4e116c1cfd83bb29acd227195d97f2ec8944452808232c310'
> +  end
> +
> +  binary_url ({
> +  })
> +  binary_sha256 ({
> +  })
> +
> +  depends_on 'gtk2'
> +  depends_on 'libgconf'
> +  depends_on 'xdg_base'
>
> Why do we need xdg_base?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/skycocker/chromebrew/pull/2529#pullrequestreview-144923344>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AlC1CU6tVmoJbuuIujD5YUj4RVc6NpYXks5uPGOLgaJpZM4V0uCP>
> .
>

uberhacker commented on packages/six.rb:
```diff
@@ -0,0 +1,22 @@
+require 'package'
+
+class Setuptools < Package
```
Dude, please don't add stuff without testing it first.  You are WAY too fast.  You need to slow down and test before spewing out random commits.  This class does not match the six.rb filename.

JL2210 replied:
Sorry. Fixed that. I'm sort of stressed to get this fixed. I have something to do tomorrow. It's done.

uberhacker replied:
Maybe we should just shelve this until you are ready?  It can wait. :)

JL2210 replied:
I think I'm good now.

uberhacker commented on packages/six.rb:
```diff
@@ -0,0 +1,22 @@
+  description 'Six is a Python 2 and 3 compatibility library.'
+  homepage 'https://github.com/benjaminp/six'
+  version '1.11.0'
+  source_url 'https://github.com/skycocker/chromebrew/raw/cc4f47007b6fe702e03de28347e20d3a0bb692f6/install.sh'
```
Why don't you grab the source from `https://github.com/benjaminp/six/archive/1.11.0.tar.gz` and run the setup.py script?

JL2210 replied:
Sure.

uberhacker commented on packages/six.rb:
```diff
@@ -0,0 +1,27 @@
+
+  def self.install
+    system "python2.7 setup.py install --prefix #{CREW_PREFIX} --root #{CREW_DEST_DIR}"
+    system "python3 setup.py install --prefix #{CREW_PREFIX} --root #{CREW_DEST_DIR}"
```
I'm glad you added this package.  There are several other packages that could be updated to use this as a dependency.  Also might be worth making a `setuptools.rb` package as well.  Something to think about.

JL2210 replied:
That's actually what I based this package off of originally. I will make another PR.

JL2210 commented:
How about 'exit'?

On Thu, Aug 9, 2018, 2:50 PM Ed Reel <notifications@github.com> wrote:

> *@uberhacker* commented on this pull request.
> ------------------------------
>
> In packages/code.rb
> <https://github.com/skycocker/chromebrew/pull/2529#discussion_r209040704>:
>
> > +  def self.install
> +    case ARCH
> +    when 'x86_64', 'i686'
> +      system "mkdir", "-p", "#{CREW_DEST_PREFIX}/share/code"
> +      system "mkdir", "-p", "#{CREW_DEST_PREFIX}/bin"
> +      system "cp", "-rpa", ".", "#{CREW_DEST_PREFIX}/share/code/"
> +      system "ln", "-s", "#{CREW_PREFIX}/share/code/bin/code", "#{CREW_DEST_PREFIX}/bin"
> +    else
> +      puts
> +      puts 'Visual Studio Code is currently not supported on ARM and AArch64.'.lightred
> +      puts 'Please try HeadMelted.'.lightred
> +      puts 'https://code.headmelted.com'.lightred
> +      puts
> +      puts 'Happy coding!'.lightgreen
> +      puts
> +      return 1
>
> You could add abort but I'm thinking it might not even be necessary. Test
> it out a bit and see what happens.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/skycocker/chromebrew/pull/2529#discussion_r209040704>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AlC1CfxdqW7gyMSPHlvpq_MMswojgf2Uks5uPIRcgaJpZM4V0uCP>
> .
>

uberhacker commented:
This might help: http://blog.honeybadger.io/how-to-exit-a-ruby-program/

JL2210 commented:
Thanks. All *```
awesome
```* now

uberhacker commented on packages/gtk_doc.rb:
```diff
@@ -29,12 +29,11 @@ class Gtk_doc < Package
   depends_on 'docbook'
   depends_on 'docbook_xml'
   depends_on 'docbook_xsl'
+  depends_on 'six'
```
Can we remove the pre-built binaries and bump the version?  Does the six package fix so this compiles cleanly now?

uberhacker replied:
Actually, there is an update available.  Upgrade to https://github.com/GNOME/gtk-doc/archive/GTK_DOC_1_28.tar.gz?

JL2210 replied:
No. Version bump won't matter, crew recursively checks dependencies anyway.
This package is cr&p.

uberhacker replied:
If this package is not needed to get code working, I would rather it be removed from this PR and submitted separately.

uberhacker replied:
Dude, you completely changed the build section.  This needs new binaries.

JL2210 replied:
Maybe. I'll try it.

uberhacker replied:
Try upgrading to version 1.28.  Always gotta think about moving forward. :)

JL2210 replied:
Nope.

JL2210 replied:
Fixed. No more binaries for docbook_xsl, docbook_xml, or docbook ever.

uberhacker replied:
I still see binaries.

JL2210 replied:
It works now. Try it.

uberhacker replied:
Dude, there is no difference between binaries and compiling from source.  Well, the difference is binary installs are a LOT faster but we're not gonna compile these each time.  That doesn't make sense.

JL2210 commented:
No. Version bump won't matter, crew recursively checks dependencies anyway.
This package is cr&p.

On Thu, Aug 9, 2018, 3:59 PM Ed Reel <notifications@github.com> wrote:

> *@uberhacker* commented on this pull request.
> ------------------------------
>
> In packages/gtk_doc.rb
> <https://github.com/skycocker/chromebrew/pull/2529#discussion_r209059709>:
>
> > @@ -29,12 +29,11 @@ class Gtk_doc < Package
>    depends_on 'docbook'
>    depends_on 'docbook_xml'
>    depends_on 'docbook_xsl'
> +  depends_on 'six'
>
> Can we remove the pre-built binaries and bump the version? Does the six
> package fix so this compiles cleanly now?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/skycocker/chromebrew/pull/2529#pullrequestreview-145001920>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AlC1CSJzjrexaVVhRoTHNAkmPitj_3qHks5uPJSrgaJpZM4V0uCP>
> .
>

uberhacker commented on packages/docbook_xsl.rb:
```diff
@@ -33,13 +26,8 @@ def self.build
-            
-            ln -s VERSION #{CREW_DEST_DIR}/usr/local/share/xml/docbook/xsl-stylesheets-1.79.1/VERSION.xsl &&
+            cp -v -R . \
+                #{CREW_DEST_DIR}/usr/local/share/xml/docbook/xsl-stylesheets-1.79.1/ &&
```
As long as we're making changes:
```
- #{CREW_DEST_DIR}/usr/local/share
+ #{CREW_DEST_PREFIX}/share
```

JL2210 commented:
@uberhacker, #{CREW_PREFIX}/etc/xml/catalog is overwritten by those binary packages. A tarball isn't smart enough to fix that.

JL2210 commented on tools/needs_binaries.sh:
```diff
@@ -7,7 +7,7 @@ cd ../packages
 exclusions="cabal.rb cros_resize.rb docx2txt.rb dpkg.rb fpc.rb freedos.rb ghc.rb kr.rb"
 exclusions+=" libtinfo.rb lsb_release.rb mysqltuner.rb nodebrew.rb nvm.rb oc.rb stack.rb"
-exclusions+=" txt2regex.rb uwsgi.rb xdg_base.rb"
+exclusions+=" txt2regex.rb uwsgi.rb xdg_base.rb docbook.rb docbook_xsl.rb docbook_xml.rb"
```
No! Why?

JL2210 commented on tools/generate_binaries.sh:
```diff
@@ -40,6 +40,9 @@ arch=$(uname -m)
 exclusions="buildessential glibc parallel wayland xorg_proto"
 
+# Exclude packages that binary packages break
+exclusions="$exclusions docbook_xsl docbook_xml docbook"
```
Why not? It's true. Installing from source is more dynamic than uncompressing binaries over a filesystem, and that's what this package needs.

uberhacker replied:
Maybe the issue is all these "docbook" packages were built at different times.  What if we make fresh binaries out of everything at the same time?  Not sure why you are so worried about the catalog.  In any event, please move these out of this PR into a separate PR.

JL2210 commented:
I changed the title. I need to get this working, or else it will break.
Currently, you have me set up to create at least 20+ PRs for all the
updates and fixes I have. I'd like it if you didn't do that.

On Thu, Aug 9, 2018, 5:25 PM Ed Reel <notifications@github.com> wrote:

> *@uberhacker* commented on this pull request.
> ------------------------------
>
> In packages/dbus_glib.rb
> <https://github.com/skycocker/chromebrew/pull/2529#discussion_r209084023>:
>
> > @@ -0,0 +1,25 @@
> +require 'package'
> +
> +class Dbus_glib < Package
> +  description 'The D-Bus GLib package contains GLib interfaces to the D-Bus API.'
> +  homepage 'http://www.linuxfromscratch.org/blfs/view/svn/general/dbus-glib.html'
> +  version '0.110'
> +  source_url 'https://dbus.freedesktop.org/releases/dbus-glib/dbus-glib-0.110.tar.gz'
> +  source_sha256 '7ce4760cf66c69148f6bd6c92feaabb8812dee30846b24cd0f7395c436d7e825'
> +
> +  binary_url ({
> +  })
> +  binary_sha256 ({
> +  })
> +
> +  depends_on 'gtk_doc'
>
> Can we not just remove the gtk_doc dependency here and submit the docbook
> madness in another PR please?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/skycocker/chromebrew/pull/2529#discussion_r209084023>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AlC1CQ35ABGG1sVq_ZLhe3TVKa0LIrThks5uPKjkgaJpZM4V0uCP>
> .
>

JL2210 commented:
Another one?

On Thu, Aug 9, 2018, 5:51 PM Ed Reel <notifications@github.com> wrote:

> *@uberhacker* commented on this pull request.
> ------------------------------
>
> In tools/generate_binaries.sh
> <https://github.com/skycocker/chromebrew/pull/2529#discussion_r209090563>:
>
> > @@ -40,6 +40,9 @@ arch=$(uname -m)
>  # Exclude fake packages
>  exclusions="buildessential glibc parallel wayland xorg_proto"
>
> +# Exclude packages that binary packages break
> +exclusions="$exclusions docbook_xsl docbook_xml docbook"
>
> Maybe the issue is all these "docbook" packages were built at different
> times. What if we make fresh binaries out of everything at the same time?
> Not sure why you are so worried about the catalog. In any event, please
> move these out of this PR into a separate PR.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/skycocker/chromebrew/pull/2529#discussion_r209090563>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AlC1CSbm98sfQ2mI4UetDyTL7zHMV-DVks5uPK77gaJpZM4V0uCP>
> .
>

uberhacker commented:
Just create a new branch and submit a new PR.  I can close this to make it easier.  The simplest solution which I mentioned 3 times now is just remove `depends_on 'gtk_doc'` from dbus_glib and call it a day..

JL2210 commented:
OK.
